universe/Pathfinder.cpp

```diff
@@ -98,102 +99,70 @@ namespace {
             m_storage.resize(a_size);
         }
 
-        /**row_lock has a mutex for the whole table and for the given row.*/
-        class row_lock {
-            private:
-            /**Table lock.*/
-            boost::shared_lock<boost::shared_mutex> m_lock;
-            /**Row lock.*/
-            boost::unique_lock<boost::shared_mutex> m_row_lock;
-
-            /**Swap the table lock with \p guard and the row lock with
-               \p row_guard.*/
-            void swap(boost::shared_lock<boost::shared_mutex>& guard, boost::unique_lock<boost::shared_mutex>& row_guard) {
-                m_lock.swap(guard);
-                m_row_lock.swap(row_guard);
-            }
-            friend class distance_matrix_cache<Storage, T, Row>;
-
-            public:
-            row_lock() {};
-            /**Swap locks with \p other row_lock.*/
-            void swap(row_lock& other) {
-                m_lock.swap(other.m_lock);
-                m_row_lock.swap(other.m_row_lock);
-            }
-            /**Unlock.*/
-            void unlock() {
-                m_row_lock.unlock();
-                m_lock.unlock();
-            }
-        };
-
         public:
-        /** try to retrieve an element, lock the whole row on cache miss
-          * if \p lock already holds a lock, it will be unlocked after locking the row.
-          */
-        boost::optional<T> get_or_lock_row(size_t row_index, size_t column_index, row_lock& lock) const {
```

Comment:
comment on line 77 refers to "get_or_row_lock" which seemingly didn't exist before, and still doesn't exist after these changes.